cpp/src/barretenberg/proof_system/circuit_builder/goblin_translator_circuit_builder.hpp

```diff
@@ -0,0 +1,700 @@
+                // current_accumulator = previous_accumulator ⋅ x + op + P.x ⋅ v + P.y ⋅ v² + z_1 ⋅ v³ + z_2 ⋅ v⁴ mod Fp
+                // To ensure this we transform the relation into the form:
+                // previous_accumulator ⋅ x + op + P.x ⋅ v + P.y ⋅ v² + z_1 ⋅ v³ + z_2 ⋅ v⁴ - quotient ⋅ p -
+                // current_accumulator = 0 However, we don't have integers. Despite that, we can approximate integers
+                // for a certain range, if we know that there will not be any overflows.
+                // For now we set the range to 2²⁷² ⋅ r. We can evaluate the logic modulo 2²⁷² with range constraints
+                // and r is native.
+                //
+                // previous_accumulator ⋅ x + op + P.x ⋅ v + P.y ⋅ v² + z_1 ⋅ v³ + z_2 ⋅ v⁴ - quotient ⋅ p -
+                // current_accumulator = 0 =>
+                // 1. previous_accumulator ⋅ x + op + P.x ⋅ v + P.y ⋅ v² + z_1 ⋅ v³ + z_2 ⋅ v⁴ + quotient ⋅ (-p mod
+                // 2²⁷²) - current_accumulator = 0 mod 2²⁷²
+                // 2. previous_accumulator ⋅ x + op + P.x ⋅ v + P.y ⋅ v² + z_1 ⋅ v³ + z_2 ⋅ v⁴ - quotient ⋅ p -
+                // current_accumulator = 0 mod r
+                //
+                // The second relation is straightforward and easy to check. The first, not so much. We have to evaluate
+                // certain bit chunks of the equation and ensure that they are zero. For example, for the lowest limb it
+                // would be (inclusive ranges):
+                //
+                // previous_accumulator[0:67] ⋅ x[0:67] + op + P.x[0:67] ⋅ v[0:67] + P.y[0:67] ⋅ v²[0:67] + z_1[0:67] ⋅
+                // v³[0:67] + z_2[0:67] ⋅ v⁴[0:67] + quotient[0:67] ⋅ (-p mod 2²⁷²)[0:67] - current_accumulator[0:67] =
+                // intermediate_value; (we don't take parts of op, because it's supposed to be between 0 and 3)
+                //
+                // We could check that this intermediate_value is equal to  0 mod 2⁶⁸ by dividing it by 2⁶⁸ and
+                // constraining it. For efficiency, we actually compute wider evaluations for 136 bits, which require us
+                // to also obtain and shift products of [68:135] by [0:67] and [0:67] by [68:135] bits.
+                // The result of division goes into the next evaluation (the same as a carry flag would)
+                // So the lowest wide limb is : (∑everything[0:67]⋅everything[0:67] +
+                // 2⁶⁸⋅(∑everything[0:67]⋅everything[68:135]))/ 2¹³⁶
+                //
+                // The high is:
+                // (low_limb + ∑everything[0:67]⋅everything[136:203] + ∑everything[68:135]⋅everything[68:135] +
+                // 2⁶⁸(∑everything[0:67]⋅everything[204:271] + ∑everything[68:135]⋅everything[136:203])) / 2¹³⁶
+                //
+                // We also limit computation on limbs of op, z_1 and z_2, since we know that op has only the lowest limb
+                // and z_1 and z_2 have only the two lowest limbs
```

Comment:
These comments are very helpful, thanks for adding them. Without any context (maybe this is all very similar to things done in bigfield) these were very hard to parse. Hopefully we will eventually have thorough documentation of how all of this works but I appreciate that that is difficult as things are in flux.

Reply:
Yes, I've been living and breathing bigfield for a few months, so there is some professional deformation. Will update when we choose the final state of limbs